tests/integration/framework/process/sqlite/sqlite.go

```diff
@@ -0,0 +1,147 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"path/filepath"
+	"strconv"
+	"testing"
+
+	// Blank import for the sqlite driver
+	_ "modernc.org/sqlite"
+
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	commonapi "github.com/dapr/dapr/pkg/apis/common"
+	componentsv1alpha1 "github.com/dapr/dapr/pkg/apis/components/v1alpha1"
+)
+
+// Option is a function that configures the process.
+type Option func(*options)
+
+// SQLite database that can be used in integration tests.
+type SQLite struct {
+	dbPath string
+	conn   *sql.DB
+	opts   options
+}
+
+func New(t *testing.T, fopts ...Option) *SQLite {
+	t.Helper()
+
+	s := &SQLite{
+		opts: options{
+			name:     "mystore",
+			metadata: map[string]string{},
+			connect:  true,
+		},
+	}
+
+	for _, fopt := range fopts {
+		fopt(&s.opts)
+	}
+
+	// Create a SQLite database in the test's temporary directory
+	tmpDir := t.TempDir()
+	s.dbPath = filepath.Join(tmpDir, "test-data.db")
+	// t.Logf("Storing SQLite database at %s", s.dbPath)
+
+	// Connect to the database
+	// We do this here because the test's init code may need to access the database before starting the test
+	if s.opts.connect {
+		var err error
+		s.conn, err = sql.Open("sqlite", "file://"+s.dbPath+"?_txlock=immediate&_pragma=busy_timeout(5000)&_pragma=journal_mode(WAL)")
+		require.NoError(t, err, "Failed to connect to SQLite database")
```

Comment:
That's why there's `s.opts.connect`. It either needs the connection for the entire test, or not. It should be fine.

Reply:
The `conn` here is a runtime test dependency and does not belong in the `New` function. Please look at other framework processes to see how `New` is used to instantiate options and configs. Runtime dependencies are created through `Run`. Ports are the only special case I can think of because of the nature of needing to both retrieve a reserved port from `0` and needing to squat that port till test execution. We don't have that requirement here.

Reply:
Ok